### ISSUE TITLE
Add documentation describing workflow design choices

### ISSUE THREAD

**Issue by scwatts** [nf-core/oncoanalyser#84]
Overview

- in order to assist accessibility to developers documentation on design choices should be made available
- this should also include other relevant information such as less used Groovy syntax
  - https://github.com/nf-core/oncoanalyser/issues/82
- it has been recommended to include this documentation in an existing file rather that a new file e.g.
  - README.md
  - docs/usage.md
  - under relevant workflow/subworkflow NF files

Initial discussion

1. https://github.com/nf-core/oncoanalyser/pull/29#discussion_r1699758567
